### ISSUE TITLE
Very low precision when using COORDINATE_SYSTEM.METER_OFFSETS

### ISSUE THREAD

**Issue by georgios-uber** [uber/deck.gl#1165]
Apply this patch to geojson demo. Then play with **origin** to see the offset. Use the **fp64** layer as reference.

```diff
diff --git a/examples/geojson/deckgl-overlay.js b/examples/geojson/deckgl-overlay.js
index c1aeedcf..0aea585e 100644
--- a/examples/geojson/deckgl-overlay.js
+++ b/examples/geojson/deckgl-overlay.js
@@ -1,5 +1,7 @@
 import React, {Component} from 'react';
-import DeckGL, {GeoJsonLayer} from 'deck.gl';
+import DeckGL, {COORDINATE_SYSTEM,ScatterplotLayer,WebMercatorViewport} from 'deck.gl';
+import {point} from '@turf/helpers';
+import turfDistance from '@turf/distance';
 
 const LIGHT_SETTINGS = {
   lightsPosition: [-125, 50.5, 5000, -122.8, 48.5, 8000],
@@ -14,11 +16,11 @@ export default class DeckGLOverlay extends Component {
 
   static get defaultViewport() {
     return {
-      latitude: 49.254,
-      longitude: -123.13,
-      zoom: 11,
-      maxZoom: 16,
-      pitch: 45,
+      latitude: 37.775516,
+      longitude: -122.418222,
+      zoom: 14,
+      maxZoom: 20,
+      pitch: 0,
       bearing: 0
     };
   }
@@ -30,25 +32,76 @@ export default class DeckGLOverlay extends Component {
       return null;
     }
 
-    const layer = new GeoJsonLayer({
-      id: 'geojson',
-      data,
-      opacity: 0.8,
-      stroked: false,
-      filled: true,
-      extruded: true,
-      wireframe: true,
+    // const layer = new GeoJsonLayer({
+    //   id: 'geojson',
+    //   data,
+    //   opacity: 0.8,
+    //   stroked: false,
+    //   filled: true,
+    //   extruded: true,
+    //   wireframe: true,
+    //   fp64: true,
+    //   getElevation: f => Math.sqrt(f.properties.valuePerSqm) * 10,
+    //   getFillColor: f => colorScale(f.properties.growth),
+    //   getLineColor: f => [255, 255, 255],
+    //   lightSettings: LIGHT_SETTINGS,
+    //   pickable: Boolean(this.props.onHover),
+    //   onHover: this.props.onHover
+    // });
+
+    const deckGlViewport = new WebMercatorViewport(viewport);
+
+    const target = [-122.418222, 37.775516];
+
+    const lfp64=new ScatterplotLayer({
+      id: 'lfp64',
+      data: [{position: target, radius: 20, color: [0, 255, 0]}],
       fp64: true,
-      getElevation: f => Math.sqrt(f.properties.valuePerSqm) * 10,
-      getFillColor: f => colorScale(f.properties.growth),
-      getLineColor: f => [255, 255, 255],
-      lightSettings: LIGHT_SETTINGS,
-      pickable: Boolean(this.props.onHover),
-      onHover: this.props.onHover
+      opacity: 0.5
+    });
+
+    // const lfp32=new ScatterplotLayer({
+    //   id: 'lfp32',
+    //   data: [{position: [-122.418222, 37.775516], radius: 20, color: [255, 0, 0]}],
+    //   fp64: false,
+    //   opacity: 0.5
+    // });
+
+    const origin = [-122, 37];
+
+    function turfDistanceX([a,b],[c,d]) {
+      return -turfDistance(point([a,b]), point([c,b]));
+    }
+
+    function turfDistanceY([a,b],[c,d]) {
+      return turfDistance(point([a,b]), point([a,d]));
+    }
+
+    const distanceX = 1000 * turfDistanceX(origin, target);
+    const distanceY = 1000 * turfDistanceY(origin, target);
+
+    const offset1 = [distanceX, distanceY];
+    const offset2 = deckGlViewport.lngLatDeltaToMeters([target[0]-origin[0], target[1]-origin[1]]);
+
+    console.log("turf>>>", distanceX, distanceY);
+    console.log("deck>>>", offset2);
+
+    const rel32=new ScatterplotLayer({
+      id: 'rel32',
+      data: [
+        {position: [0,0], radius: 200, color: [0, 0, 0]},
+        {position: offset1, radius: 10, color: [0, 0, 255]},
+        {position: offset2, radius: 10, color: [255, 0, 0]}
+      ],
+      fp64: false,
+      opacity: 0.5,
+
+      coordinateSystem: COORDINATE_SYSTEM.METER_OFFSETS,
+      coordinateOrigin: origin
     });
 
     return (
-      <DeckGL {...viewport} layers={ [layer] } initWebGLParameters />
+      <DeckGL {...viewport} layers={ [lfp64, rel32] } initWebGLParameters />
     );
   }
 }

```

**Comment by 1chandu** [comment]
@georgios-uber , having build issues : 
`Module not found: Error: Can't resolve '@turf/distance'`

Can you push the changes to a remote branch, and also what you mean by `play with origin to see the offset` , does the offset has big difference between `fp32` and `fp64` modes?

**Comment by georgios-uber** [comment]
@1chandu You can

> yarn add @turf/turf

**Comment by georgios-uber** [comment]
@1chandu pushed to METER_OFFSETS_PROBLEM_DEMO. See https://github.com/uber/deck.gl/commit/5d4c06e875997343ef06f03131f5725968429972


<img width="726" alt="fp64" src="https://user-images.githubusercontent.com/25012840/32969927-c58847a6-cb9b-11e7-9f4c-59cebd60d62c.png">

**Comment by georgios-uber** [comment]
fp64 does not make a difference for the relative layer.

**Comment by georgios-uber** [comment]
I have found a workaround in calculating the offsets, here is the code I use:

```
import ViewportMercator, {
  projectFlat,
  getMercatorDistanceScales
} from 'viewport-mercator-project';
const PROJECT_SCALE = 2 ** 20;

    this.lngLat = [longitude, latitude];
    this.lngLatProjectFlat = projectFlat(this.lngLat, PROJECT_SCALE);

  coordsToMeterOffset(lngLat: number[]): [number, number] {
    const pixelsLngLat = projectFlat(lngLat, PROJECT_SCALE);
    const pixels = [
      this.lngLatProjectFlat[0] - pixelsLngLat[0],
      this.lngLatProjectFlat[1] - pixelsLngLat[1]
    ];
    const { metersPerPixel } = getMercatorDistanceScales({
      latitude: lngLat[1],
      longitude: lngLat[0],
      scale: PROJECT_SCALE
    });

    return [-pixels[0] * metersPerPixel[0], pixels[1] * metersPerPixel[1]];
  }
```
This works well. Maybe it should be included as utility function in viewport-mercator-project?

**Comment by 1chandu** [comment]
@ib @shaojingli , given this is not a blocker moving this out of 5.0.

**Comment by ib** [comment]
Hi, here is __ib__. Could you please use @ibgreen instead of @ib to address __ibgreen__, so that the right person will be notified. Thank you very much.

**Comment by ibgreen** [comment]
I think this was fixed in 5.0, we should test again.
